src/services/SelfTestCheck.ts

```diff
@@ -20,12 +20,13 @@ export default class SelfTestChecker {
           throw Boom.badImplementation('self test stats missing from plugin context')
         }
 
```

Comment:
I feel like this should be simpler to read and immediately continue if selftestSuccess is true. 

Instead of this flow what about checking the stats.selfTestSuccess immediately so it would look like:
```javascript
const stats = selfTest.getTestStats()
if (stats.selfTestSuccess) { 
	return h.continue
} 
...
```
Then the Logic as you have it updated. However 
 - Don't make a log when the header is not present. (It logs too much then)
 - Just check if the header is there, if is, grab the token. If its not respond with the self test stats and since you checked it above and thats the fall through if it has not passed.